tests/test_file_storage.py

```diff
@@ -0,0 +1,407 @@
+        (
+            file_storage(provider=FileStorageProvider.GCS),
+            TEST_CONSTANTS.READ_PDF_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_PDF_FILE,
+            "wb",
+            -1,
+            os.path.getsize(TEST_CONSTANTS.READ_PDF_FILE),
+        ),
+        (
+            file_storage(provider=FileStorageProvider.GCS),
+            TEST_CONSTANTS.READ_TEXT_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "wb",
+            0,
+            0,
+        ),
+        (
+            file_storage(provider=FileStorageProvider.GCS),
+            TEST_CONSTANTS.READ_TEXT_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "wb",
+            0,
+            0,
+        ),
+        (
+            file_storage(provider=FileStorageProvider.Local),
+            TEST_CONSTANTS.READ_PDF_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_PDF_FILE,
+            "wb",
+            -1,
+            os.path.getsize(TEST_CONSTANTS.READ_PDF_FILE),
+        ),
+        (
+            file_storage(provider=FileStorageProvider.Local),
+            TEST_CONSTANTS.READ_TEXT_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "wb",
+            -1,
+            os.path.getsize(TEST_CONSTANTS.READ_TEXT_FILE),
+        ),
+        (
+            file_storage(provider=FileStorageProvider.Local),
+            TEST_CONSTANTS.READ_TEXT_FILE,
+            "rb",
+            None,
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "wb",
+            0,
+            0,
+        ),
+        (
+            file_storage(provider=FileStorageProvider.GCS),
+            None,
+            "rb",
+            TEST_CONSTANTS.TEXT_CONTENT,
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "w",
+            0,
+            len(TEST_CONSTANTS.TEXT_CONTENT),
+        ),
+        (
+            file_storage(provider=FileStorageProvider.Local),
+            None,
+            "rb",
+            TEST_CONSTANTS.TEXT_CONTENT.encode(),
+            TEST_CONSTANTS.WRITE_TEXT_FILE,
+            "wb",
+            0,
+            len(TEST_CONSTANTS.TEXT_CONTENT.encode()),
+        ),
+    ],
+)
+def test_file_write(
+    file_storage,
+    read_file_path,
+    read_mode,
+    file_contents,
+    write_file_path,
+    write_mode,
+    read_length,
+    expected_write_length,
+):
+    if file_storage.exists(path=TEST_CONSTANTS.WRITE_FOLDER_PATH):
+        file_storage.rm(path=TEST_CONSTANTS.WRITE_FOLDER_PATH, recursive=True)
+    assert file_storage.exists(path=TEST_CONSTANTS.WRITE_FOLDER_PATH) == False  # noqa
+    file_storage.mkdir(path=TEST_CONSTANTS.WRITE_FOLDER_PATH)
+    # assert file_storage.exists(path=TEST_CONSTANTS.WRITE_FOLDER_PATH) == True
+    if read_file_path:
+        file_read_handle = file_storage.open(path=read_file_path, mode=read_mode)
```

Comment:
NIT: The file handle might not be closed here since we need to explicitly close it

Reply:
Comment is outdated as APIs have changed.